sqrl-server/sqrl-server-vertx/src/main/java/com/datasqrl/graphql/GraphQLServer.java

```diff
@@ -195,17 +198,41 @@ private void logRequests(Router router) {
     router.route().handler(LoggerHandler.create());
   }
 
-  private void configureGraphiQLHandler(Router router) {
-    if (this.config.getGraphiQLHandlerOptions() != null) {
+  private void configureGraphiQLEndpoint(Router router, Vertx vertx) {
+    if (config.getGraphiQLHandlerOptions() != null) {
+      GraphiQLHandlerBuilder handlerBuilder = GraphiQLHandler.builder(vertx)
+          .with(config.getGraphiQLHandlerOptions());
+
+      if (authProvider != null) {
+        String token = authProvider.generateToken(new JsonObject());
+
+        handlerBuilder.addingHeaders(rc ->
+            MultiMap.caseInsensitiveMultiMap().add("Authorization", "Bearer " + token));
+      }
+
       router.route(config.getServletConfig().getGraphiQLEndpoint())
-          .handler(GraphiQLHandler.create(config.getGraphiQLHandlerOptions()));
+          .subRouter(handlerBuilder.build().router());
     }
   }
 
   private void handleErrors(Router router) {
     router.errorHandler(500, ctx -> {
-      log.error("", ctx.failure());
-      ctx.response().setStatusCode(500).end();
+      log.error(String.format(
+          "An internal error happened during processing the request. request.uri: [%s]",
+          ctx.request().uri()
+      ), ctx.failure());
+      ctx.response().setStatusCode(500).end(
+          "{ \"statusCode\": 500, \"statusMessage\": \"Internal Server Error\" }"
+      );
+    });
+    router.errorHandler(401, ctx -> {
+      log.error(String.format(
+          "Unauthorized. request.uri: [%s]",
+          ctx.request().uri()
+      ), ctx.failure());
+      ctx.response().setStatusCode(401).end(
+          "{ \"statusCode\": 401, \"statusMessage\": \"Unauthorized\" }"
```

Comment:
Normally, a JSON body as a response is not required when the status code is 401. However, GraphiQL does require it, which is why we're seeing the unusual error message on the UI. This message is a result of the JS code being unable to parse the response body...